types/react-widgets/lib/DateTimePicker.d.ts

```diff
@@ -88,7 +88,7 @@ declare namespace DateTimePicker {
          * parsing yourself. When parse is unspecified and the format prop is a String parse will
          * automatically use that format as its default
          */
-        parse?: ((str: string) => Date) | string[] | string;
+        parse?: ((str: string) => Date | undefined) | string[] | string;
```

Comment:
This is the real bug fix: the parse callback should be able to return `undefined`